crates/rpc/src/middleware/versioning.rs

```diff
@@ -1,527 +0,0 @@
-            .body(
-                code.canonical_reason()
-                    .expect("canonical reason is defined")
-                    .into(),
-            )
-            .expect("response is properly formed")
-    }
-
-    fn with_body(code: StatusCode, body: impl Into<Body>) -> Response<Body> {
-        Builder::new()
-            .status(code)
-            .header(CONTENT_TYPE, TEXT)
-            .body(body.into())
-            .expect("response is properly formed")
-    }
-}
-
-pub mod test_utils {
-    pub mod method_names {
-        pub const COMMON_FOR_V03_V04: [&str; 23] = [
-            "starknet_addDeclareTransaction",
-            "starknet_addDeployAccountTransaction",
-            "starknet_addInvokeTransaction",
-            "starknet_blockHashAndNumber",
-            "starknet_blockNumber",
-            "starknet_call",
-            "starknet_chainId",
-            "starknet_estimateFee",
-            "starknet_getBlockWithTxHashes",
-            "starknet_getBlockWithTxs",
-            "starknet_getBlockTransactionCount",
-            "starknet_getClass",
-            "starknet_getClassAt",
-            "starknet_getClassHashAt",
-            "starknet_getEvents",
-            "starknet_getNonce",
-            "starknet_getStateUpdate",
-            "starknet_getStorageAt",
-            "starknet_getTransactionByBlockIdAndIndex",
-            "starknet_getTransactionByHash",
-            "starknet_getTransactionReceipt",
-            "starknet_pendingTransactions",
-            "starknet_syncing",
-        ];
-        pub const COMMON_FOR_ALL: [&str; 2] =
-            ["pathfinder_getProof", "pathfinder_getTransactionStatus"];
-        pub const V03_ONLY: [&str; 1] = ["starknet_simulateTransaction"];
-        pub const V04_ONLY: [&str; 1] = ["starknet_simulateTransactions"];
-        pub const PATHFINDER_ONLY: [&str; 1] = ["pathfinder_version"];
-    }
-
-    pub mod paths {
-        pub const ROOT: &[&str] = &["", "/"];
-        pub const V03: &[&str] = &["/rpc/v0.3", "/rpc/v0.3/"];
-        pub const V04: &[&str] = &["/rpc/v0.4", "/rpc/v0.4/"];
-        pub const PATHFINDER: &[&str] = &["/rpc/pathfinder/v0.1", "/rpc/pathfinder/v0.1/"];
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use super::test_utils::{method_names, paths};
-    use super::{prefix_rpc_method_names_with_version, DefaultVersion};
-    use crate::test_client::TestClientBuilder;
-    use crate::{RpcContext, RpcServer};
-    use http::Method;
-    use jsonrpsee::core::error::Error;
-    use jsonrpsee::types::error::{CallError, METHOD_NOT_FOUND_CODE};
-    use rstest::rstest;
-    use serde_json::json;
-
-    // In an unintentional way OFC: if a method is INTENDED to be available
-    // on many paths then this is absolutely allowed.
-    #[rstest]
-    #[case(DefaultVersion::V03)]
-    #[case(DefaultVersion::V04)]
-    #[tokio::test]
-    async fn api_versions_dont_leak_between_each_other(#[case] default_version: DefaultVersion) {
```

Comment:
I think that this test should be retained as it is complementary to the integration test (`rpc/tests/versioning.rs`) which checks the opposite scenario. Apart from those two tests there's no test to check that registration worked for all methods it was intended to.

Reply:
agreed - not needed